lms/services/application_instance.py

```diff
@@ -165,6 +167,7 @@ def search(
             .all()
         )
 
+    # pylint: disable=too-complex
```

Comment:
I reckon this might have triggered you to do the other refactor around search, how dare you pylint, you are too complex!

Reply:
No comment